source/code/go/src/plugins/telemetry.go

```diff
@@ -159,6 +161,25 @@ func InitializeTelemetryClient(agentVersion string) (int, error) {
 		Log("Overriding the default AppInsights EndpointUrl with %s", appInsightsEndpoint)
 		telemetryClientConfig.EndpointUrl = envAppInsightsEndpoint
 	}
+	// if the proxy configured set the customized httpclient with proxy
+	isProxyConfigured := false
+	if ProxyEndpoint != "" {
```

Comment:
So you are checking if proxy is a valid uri here ? This is too late. Ideally you should validate proxy uri in main.sh (which is the entry point), and set the env variable if its valid and everybody should just read from env var & use it. If its invalid, just dont set the env var, and things will fail as it is before.

Reply:
Done.